integrations/qdrant/tests/test_retriever.py

```diff
@@ -112,3 +117,126 @@ def test_run(self, filterable_docs: List[Document]):
+                        "replication_factor": None,
+                        "write_consistency_factor": None,
+                        "on_disk_payload": None,
+                        "hnsw_config": None,
+                        "optimizers_config": None,
+                        "wal_config": None,
+                        "quantization_config": None,
+                        "init_from": None,
+                        "wait_result_from_api": True,
+                        "metadata": {},
+                        "write_batch_size": 100,
+                        "scroll_size": 10000,
+                        "payload_fields_to_index": None,
+                    },
+                },
+                "filters": None,
+                "top_k": 10,
+                "scale_score": True,
+                "return_embedding": False,
+            },
+        }
+
+    def test_from_dict(self):
+        data = {
+            "type": "haystack_integrations.components.retrievers.qdrant.retriever.QdrantSparseRetriever",
+            "init_parameters": {
+                "document_store": {
+                    "init_parameters": {"location": ":memory:", "index": "test"},
+                    "type": "haystack_integrations.document_stores.qdrant.document_store.QdrantDocumentStore",
+                },
+                "filters": None,
+                "top_k": 5,
+                "scale_score": False,
+                "return_embedding": True,
+            },
+        }
+        retriever = QdrantSparseRetriever.from_dict(data)
+        assert isinstance(retriever._document_store, QdrantDocumentStore)
+        assert retriever._document_store.index == "test"
+        assert retriever._filters is None
+        assert retriever._top_k == 5
+        assert retriever._scale_score is False
+        assert retriever._return_embedding is True
+
+    def _generate_mocked_sparse_embedding(self, n):
+        list_of_sparse_vectors = []
+        for _ in range(n):
+            random_indice_length = np.random.randint(3, 15)
+            data = {
+                "indices": list(range(random_indice_length)),
+                "values": [np.random.random_sample() for _ in range(random_indice_length)],
+            }
+            list_of_sparse_vectors.append(data)
+        return list_of_sparse_vectors
+
+    def test_run(self, filterable_docs: List[Document]):
+        document_store = QdrantDocumentStore(location=":memory:", index="Boi")
+
+        # Add fake sparse embedding to documents
+        for doc in filterable_docs:
+            doc.sparse_embedding = SparseEmbedding.from_dict(self._generate_mocked_sparse_embedding(1)[0])
+
+        document_store.write_documents(filterable_docs)
+
+        retriever = QdrantSparseRetriever(document_store=document_store)
+
+        results: List[Document] = retriever.run(query_sparse_embedding=self._generate_mocked_sparse_embedding(1)[0])
+
+        assert len(results["documents"]) == 10  # type: ignore
```

Comment:
why type:ignore?

Reply:
I used the same template as previous retriever (dense)
Without:
```python
cmd [3] | mypy --install-types --non-interactive --explicit-package-bases src/ tests
tests/test_retriever.py:233: error: No overload variant of "__getitem__" of "list" matches argument type "str"  [call-overload]
tests/test_retriever.py:233: note: Possible overload variants:
tests/test_retriever.py:233: note:     def __getitem__(self, SupportsIndex, /) -> Any
tests/test_retriever.py:233: note:     def __getitem__(self, slice, /) -> list[Any]
Found 1 error in 1 file (checked 13 source files)

```